azure/datalake/store/lib.py

```diff
@@ -141,23 +143,43 @@ def auth(tenant_id=None, username=None,
                                                             client_secret)
     else:
         raise ValueError("No authentication method found for credentials")
+
     out.update({'access': out['accessToken'], 'resource': resource,
                 'refresh': out.get('refreshToken', False),
                 'time': time.time(), 'tenant': tenant_id, 'client': client_id})
-    return out
 
+    return DataLakeCredential(out)
+
+class DataLakeCredential(Authentication):
```

Comment:
We need to discuss that, I wasn't able to understand your issue in the email thread. Why do you need here something that is supposed to be generic? Or you have some ADL behavior that requires a specific class?